### PR TITLE
fix(field_resolve): two-stage i64-first parsing eliminates boundary saturation (closes #421)

### LINKED ISSUE #421: feat(field_resolve): tighten i64 bounds check in parsed_number_to_wire_value (precision edge case)
## Background

`src/cli/issue/field_resolve.rs::parsed_number_to_wire_value` uses the predicate `parsed.fract() == 0.0 && parsed >= i64::MIN as f64 && parsed <= i64::MAX as f64` to decide whether to emit an i64 wire form or fall back to f64. This is **pre-existing behavior** carried forward from PR #401 (S-396) and was extracted byte-identically into a helper by S-409 (PR #418).

Surfaced as a Copilot review finding on PR #418.

## Bug

`i64::MAX as f64` rounds **up** to `9223372036854775808.0` (i.e., `2^63`) because f64 cannot exactly represent every integer above `2^53`. `i64::MAX` itself is `2^63 - 1 = 9223372036854775807`, which f64 cannot represent.

Since Rust 1.45, `f64 as i64` is saturating: out-of-range values clamp to the destination bound rather than triggering UB.

Consequence: an input string like `"9223372036854775808"` parses to f64 `9223372036854775808.0`, passes the `parsed <= i64::MAX as f64` check (because both sides are exactly `9223372036854775808.0` after rounding), and the subsequent `as i64` cast saturates silently to `i64::MAX` (`9223372036854775807`). The user sees a JSON wire value 1 unit different from their input, with no error.

Perplexity-validated source: https://rust.docs.kernel.org/core/primitive.f64.html + Rust 1.45 release notes.

## Real-world impact

Tiny. The Jira `number` custom-field schema doesn't reach 2^63 in practical use (most numeric fields are bounded much smaller). This edge case only triggers for inputs within `ε` of 2^63 (≈ 9.2 × 10^18). Has not been observed in production.

## Recommended scope

Option A (minimal): tighten the predicate to use strict-less-than against a safe upper bound:

```rust
const I64_MAX_AS_F64_SAFE: f64 = 9223372036854775000.0; // largest f64 below 2^63 - 512
if parsed.fract() == 0.0 && parsed >= i64::MIN as f64 && parsed < I64_MAX_AS_F64_SAFE {
    serde_json::Value::Number(serde_json::Number::from(parsed as i64))
} else { ... }
```

Option B (preferred): parse the original `value: &str` as i64 first; fall back to f64 only on parse failure.

```rust
if let Ok(n) = value.parse::<i64>() {
    serde_json::Value::Number(serde_json::Number::from(n))
} else {
    let parsed: f64 = value.parse()?;
    // existing NaN/Inf guard + f64 emission
}
```

Option B is cleaner — eliminates the f64-roundtrip precision loss entirely for whole-number inputs, and the i64 parser will naturally reject `"9223372036854775808"` as out of range (returning an error → fall back to f64 branch which is correct).

## Add coverage

A unit test pinning the bug: input `"9223372036854775808"` (i.e., i64::MAX + 1) MUST either (a) emit an f64 wire form, or (b) return a user-error. It must NOT silently emit `i64::MAX` (the saturated value).

## Severity

LOW (real-world impact is tiny; pre-existing, not a regression). Suitable for a maintenance sweep.

## Source

- Copilot review on PR #418 (S-409): https://github.com/Zious11/jira-cli/pull/418
- Perplexity-validated 2026-05-27 against Rust language reference + f64 docs
- Pre-existing in code since S-396 (PR #401)

### PR DESCRIPTION
**Pull request by Zious11**
## Summary

- **Bug fix:** `parsed_number_to_wire_value` silently saturated i64 boundary inputs (e.g., `"9223372036854775808"` = i64::MAX + 1) to ±i64::MAX/MIN due to f64→i64 cast via non-exact f64 representation. Surfaced by Copilot review on PR #418 (S-409) and Perplexity-validated.
- **Fix (Option C from F1 delta analysis):** Two-stage parser — Stage 1 tries `value.parse::<i64>()` for exact integers with no f64 round-trip; Stage 2 falls back to f64 with strict-inequality bounds (`>` / `<` instead of `>=` / `<=`) to correctly route out-of-range integers to f64 wire form.
- **Symmetric lower-bound fix:** `"-9223372036854775809"` (i64::MIN - 1) is now correctly emitted as f64 rather than saturated to i64::MIN.
- **Bonus precision win:** Integers above 2^53 (e.g., `"9007199254740993"`) which previously round-tripped through f64 with off-by-one errors now go through Stage 1 i64-first parse with exact representation.
- **8 new unit tests** covering: i64::MAX, i64::MAX+1, i64::MIN, i64::MIN-1, 2^53, 2^53+1, 1e10 scientific notation, and the helper predicate directly at both bounds.
- **6 existing S-409 helper unit tests + integration tests 26/27 unchanged** — behavior is identical for all previously-correct inputs.
- **F2 spec evolution** (BC-3.4.015 invariant 5 + EC-3.4.015-4b) landed separately at factory-artifacts commit `6680de7` — not part of this PR.

Closes #421

## Architecture Changes

Only `src/cli/issue/field_resolve.rs` is modified — no module additions, no new public exports, no dependency changes.

```mermaid
graph TD
    A["issue edit --field NAME=VALUE"] --> B["resolve_edit_fields()"]
    B --> C["number arm"]
    C --> D{Stage 1: value.parse::<i64>()}
    D -->|Ok| E["serde_json::Number::from(n)\n(exact, no f64)"]
    D -->|Err| F["value.parse::<f64>()"]
    F --> G{"NaN/Inf check"}
    G -->|not finite| H["JrError::InvalidArgument"]
    G -->|finite| I["parsed_number_to_wire_value(parsed)"]
    I --> J{fract==0 AND\nparsed > i64::MIN as f64\nAND parsed < i64::MAX as f64}
    J -->|true| K["serde_json::Number from i64 cast"]
    J -->|false| L["serde_json::Number from f64"]
```

## Story Dependencies

```mermaid
graph LR
    S409["S-409: extract parsed_number_to_wire_value\n(PR #418, merged)"] --> S421["S-421: fix i64 boundary saturation\n(this PR)"]
    S396["S-396: issue edit --field"] --> S409
```

No story depends on S-421 (leaf node).

## Spec Traceability

```mermaid
flowchart LR
    BC["BC-3.4.015\n(number field wire form)"] --> AC1["AC-001\ntwo-stage call site"]
    BC --> AC2["AC-002\nstrict inequality helper"]
    BC --> AC3["AC-003\n8 new unit tests"]
    BC --> AC4["AC-004\nS-409 tests unchanged"]
    BC --> AC5["AC-005\nInteg tests 26/27 pass"]
    BC --> AC6["AC-006\ncargo test exits 0"]
    BC --> AC7["AC-007\nlint+fmt clean"]
    BC --> AC8["AC-008\nF2 BC changes verified"]
    AC1 --> T1["test_parse_number_i64_max_boundary\ntest_parse_number_i64_max_plus_one_is_f64"]
    AC2 --> T2["test_parsed_number_to_wire_value_strict_upper\ntest_parsed_number_to_wire_value_strict_lower"]
    AC3 --> T3["8 boundary unit tests\n(field_resolve.rs::tests)"]
    AC4 --> T4["6 S-409 tests\n(unchanged)"]
    AC5 --> T5["test_bc_3_4_015_number_field_integer_wire_form\ntest_bc_3_4_015_number_field_scientific_notation_wire_form"]
    T1 --> Impl["field_resolve.rs\nStage 1 + Stage 2 + strict predicate"]
    T2 --> Impl
    T3 --> Impl
    T4 --> Impl
    T5 --> Impl
```

## Test Evidence

| Suite | Count | Result |
|-------|-------|--------|
| `cargo test --lib field_resolve` | 14 (6 S-409 + 8 new) | PASS |
| `cargo test --test issue_edit_field` | 54 (tests 26/27 green) | PASS |
| `cargo test` full suite | all | PASS (see CI) |
| `cargo fmt --all -- --check` | — | PASS |
| `cargo clippy --all-targets -- -D warnings` | — | PASS |

**New tests by AC:**
- AC-001/003: `test_parse_number_i64_max_boundary`, `test_parse_number_i64_max_plus_one_is_f64`, `test_parse_number_i64_min_boundary`, `test_parse_number_i64_min_minus_one_is_f64`, `test_parse_number_2_to_53_exact_f64`, `test_parse_number_2_to_53_plus_one_exact_i64`, `test_parse_number_scientific_1e10_is_i64`
- AC-002: `test_parsed_number_to_wire_value_strict_upper_excludes_two_to_the_63`, `test_parsed_number_to_wire_value_strict_lower_excludes_negative_two_to_the_63_in_stage2` (1 bonus test beyond spec minimum)

**Integration test regression pins (unchanged):**
- `test_bc_3_4_015_number_field_integer_wire_form` (test 26, input `"5"` → Stage 1 → JSON int `5`)
- `test_bc_3_4_015_number_field_scientific_notation_wire_form` (test 27, input `"5e3"` → Stage 2 → JSON int `5000`)

## Holdout Evaluation

N/A — evaluated at wave gate.

## Adversarial Review

N/A — evaluated at Phase 5.

## Security Review

No security-sensitive changes. This PR modifies only number-field string parsing logic in `field_resolve.rs`. No new HTTP calls, no new credential handling, no new user-input deserialization paths beyond the existing `value.parse::<i64>()` and `value.parse::<f64>()` which cannot panic (they return `Result`). OWASP injection vectors: not applicable to internal type coercion. No `unsafe` code added. Security review: PASS.

## Risk Assessment

| Dimension | Rating | Notes |
|-----------|--------|-------|
| Blast radius | Narrow | Single function in one file; only number-type `--field` values affected |
| Behavioral regression risk | Low | All inputs within i64 range: identical wire form via Stage 1; Stage 2 behavior identical for decimals/scientific/f64-range-integers |
| Breaking change | No | Wire form changes only for previously-incorrect boundary inputs |
| Performance | Negligible | One additional `str::parse::<i64>()` attempt per number-field call |

## AI Pipeline Metadata

| Field | Value |
|-------|-------|
| Story | S-421 |
| Wave | feature-followup |
| Pipeline mode | TDD strict |
| Models used | claude-sonnet-4-6 |
| Phase | F3 (incremental story) |
| F1 delta analysis | `.factory/phase-f1-delta-analysis/issue-421/delta-analysis.md` |
| F2 spec evolution | factory-artifacts `6680de7` (BC-3.4.015 invariant 5 + EC-3.4.015-4b) |

## Pre-Merge Checklist

- [x] PR description matches actual diff
- [x] All 8 ACs covered by unit tests or verified green
- [x] Traceability chain complete: BC-3.4.015 → AC-001 through AC-008 → tests → implementation
- [x] No breaking changes
- [x] `cargo test` passes (14 lib + 54 integration + full suite)
- [x] `cargo fmt --all -- --check` exits 0
- [x] `cargo clippy --all-targets -- -D warnings` exits 0
- [x] No `#[allow]` lint suppressions added
- [x] No new dependencies
- [x] Integration tests 26/27 unchanged and passing
- [x] F2 BC changes verified landed (AC-008)
- [ ] CI green (pending)
- [ ] PR reviewer approved (pending)

### CODE REVIEW

File: src/cli/issue/field_resolve.rs [hunk 1]
```diff
@@ -11,17 +11,43 @@ use crate::error::JrError;
 /// and avoids implicit precision warnings on some Jira instances). Decimal values
 /// stay as f64 via `serde_json::json!`.
 ///
+/// # Context: Stage 2 (f64 fallback) caller contract
+///
+/// This helper is called from a Stage 2 (f64) fallback context — Stage 1 (i64-first
+/// parse via `value.parse::<i64>()`) at the call site has already eliminated all
+/// exact-i64-representable inputs. Inputs that reach this helper therefore came from
+/// strings that failed `parse::<i64>()`: decimals, scientific notation, and integers
+/// outside the i64 range.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit 2fa81dd: rustdoc rewritten to make clear Stage 1 only catches string forms that parse cleanly as i64 (e.g., "5", "-9223372036854775807"). Stage 2 receives decimal forms ("5.5", "5.0"), scientific notation ("5e3", "1e20"), integer strings outside i64 range, and any other inputs that failed i64 parsing. The doc now enumerates the categories explicitly with example inputs. Thanks for catching the imprecision.

File: src/cli/issue/field_resolve.rs [hunk 2]
```diff
@@ -11,17 +11,43 @@ use crate::error::JrError;
...
+/// - **Upper bound:** `i64::MAX` is 9_223_372_036_854_775_807, which is NOT exactly
+///   representable as f64 (f64 has 53-bit mantissa; integers above 2^53 are rounded).
+///   `i64::MAX as f64` rounds UP to 9_223_372_036_854_775_808.0 (= 2^63). The
+///   non-strict `<=` predicate would admit this value; `parsed as i64` then saturates
+///   silently to `i64::MAX`, producing wrong output. Strict `<` excludes 2^63.
+///
+/// - **Lower bound:** `i64::MIN as f64` is -9_223_372_036_854_775_808.0 (= -2^63),
+///   which IS exactly representable as f64. In Stage 2 context, this value can only
+///   arrive from an underflowing string like "-9223372036854775809" (Stage 1 would
+///   have consumed the exact `i64::MIN` string "-9223372036854775808" before reaching
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit 2fa81dd: comment rewritten to enumerate the three string forms that can produce parsed=-2^63 in Stage 2: (a) underflowing integer like "-9223372036854775809", (b) decimal form of i64::MIN "-9223372036854775808.0", (c) scientific notation "-9.223372036854776e18". The comment now notes that emitting f64 for cases (b)/(c) is a deliberate trade-off (loses no precision since both wire forms encode -9223372036854775808 exactly per JSON Number spec) made to keep the predicate strict and avoid the saturation bug in case (a).

File: src/cli/issue/field_resolve.rs [hunk 3]
```diff
@@ -11,17 +11,64 @@ use crate::error::JrError;
...
 pub(crate) fn parsed_number_to_wire_value(parsed: f64) -> serde_json::Value {
     debug_assert!(
         parsed.is_finite(),
         "parsed_number_to_wire_value requires a finite value; caller must reject NaN/Inf"
     );
-    if parsed.fract() == 0.0 && parsed >= i64::MIN as f64 && parsed <= i64::MAX as f64 {
+    if parsed.fract() == 0.0 && parsed > (i64::MIN as f64) && parsed < (i64::MAX as f64) {
         serde_json::Value::Number(serde_json::Number::from(parsed as i64))
     } else {
         serde_json::json!(parsed)
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs:99]: Confirmed via empirical Rust test: all four boundary strings ("9223372036854775807", "9223372036854775808", "9223372036854775807.0", "9223372036854775808.0") parse to the same f64 value 2^63 (= 9223372036854775808.0, bits 0x43e0000000000000). Your finding is correct — the strict `<` predicate I added regressed on `"9223372036854775807.0"` (decimal form of i64::MAX): pre-fix, accidental saturation+rounding cancellation gave the correct value; post-fix, strict `<` routed it to f64 wire form encoding 2^63, off-by-one HIGH from user input.\n\nFixed in commit 97e1850 by adding **Stage 1.5** between Stages 1 and 2: when Stage 1 i64-parse fails AND the input matches `^[-+]?\d+\.0+$` (integer with trailing-zero decimal suffix), strip the suffix and retry as i64. This recovers exact i64 semantics for `"5.0"`, `"9223372036854775807.0"`, `"-9223372036854775808.0"`, etc. Genuine overflows like `"9223372036854775808.0"` strip to `"9223372036854775808"`, fail the retry, and fall through to Stage 2 → emit f64 (correct, no saturation).\n\nNew helper `strip_integer_decimal_suffix` is hand-rolled (no `regex` dep). 3 new regression-pin tests + 3 new helper unit tests added (20 total in `field_resolve.rs::tests` now).

File: src/cli/issue/field_resolve.rs [hunk 4]
```diff
@@ -292,23 +339,33 @@ pub(crate) async fn resolve_edit_fields(
...
+                // pre-fix when parsed through f64 first).
+                if let Ok(n) = value.parse::<i64>() {
+                    wire_value = serde_json::Value::Number(serde_json::Number::from(n));
+                } else {
+                    // Stage 2: f64 fallback for decimals, scientific notation, and
+                    // integers outside the i64 range.
+                    let parsed: f64 = value.parse().map_err(|_| {
+                        JrError::UserError(format!(
+                            "Cannot parse '{value}' as a number for field '{human_name}'. \
+                             Provide a valid numeric value (integer or decimal)."
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit 97e1850: both occurrences of the parse-error message (Stage 1.5 fallback path and Stage 2 path) updated from `"integer or decimal"` to `"integer, decimal, or scientific notation like 1e10"`. Matches the actually-accepted input shapes.

File: src/cli/issue/field_resolve.rs [hunk 5]
```diff
@@ -496,4 +553,125 @@ mod tests {
...
+    #[test]
+    fn test_s421_i64_max_emits_i64() {
+        // Simulate Stage 1 + Stage 2 logic for "9223372036854775807"
+        let value = "9223372036854775807";
+        let wire = if let Ok(n) = value.parse::<i64>() {
+            serde_json::Value::Number(serde_json::Number::from(n))
+        } else {
+            let parsed: f64 = value.parse().unwrap();
+            parsed_number_to_wire_value(parsed)
+        };
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit 97e1850: extracted `parse_number_wire(value: &str) -> serde_json::Value` test-only helper in the `#[cfg(test)] mod tests` block. All 6 `test_s421_*` end-to-end tests now call it instead of inlining the Stage 1 / 1.5 / 2 dispatch. The helper has a rustdoc note that it must be kept in sync with the production code in `resolve_edit_fields`. The 2 helper-only tests (`test_parsed_number_to_wire_value_strict_*`) keep their direct calls since they test the helper in isolation.

File: src/cli/issue/field_resolve.rs [hunk 6]
```diff
@@ -496,4 +613,193 @@ mod tests {
...
+            wire.is_f64(),
+            "expected f64 for 2^63 (out of i64 range), got {wire}"
+        );
+    }
+
+    #[test]
+    fn test_parsed_number_to_wire_value_strict_lower_excludes_negative_two_to_the_63_in_stage2() {
+        // -2^63 = i64::MIN as f64 (exact). In Stage 2 context, this value can only
+        // arrive from an underflowing string like "-9223372036854775809"; the strict
+        // > i64::MIN comparison correctly routes it to f64.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit f7ef2fe: the test comment for `test_parsed_number_to_wire_value_strict_lower_excludes_negative_two_to_the_63_in_stage2` was carrying the old incomplete claim from R3. Updated to enumerate all three Stage-2 source cases for -2^63: underflowing integer (`"-9223372036854775809"`), decimal form of i64::MIN (`"-9223372036854775808.0"`), and scientific notation (`"-9.223372036854776e18"`). Comment now matches the (correct) rustdoc.

File: src/cli/issue/field_resolve.rs [hunk 7]
```diff
@@ -11,23 +11,94 @@ use crate::error::JrError;
...
+///     rejects it (decimal point). Value IS valid `i64::MIN`; strict `>` still routes
+///     it to f64.
+///   - (c) Scientific notation: `"-9.223372036854776e18"` — Stage 1 rejects it (`e`
+///     present). Value IS valid `i64::MIN` (approximately); strict `>` routes to f64.
+///
+///   For cases (b) and (c) the value is actually valid `i64::MIN`; emitting f64 instead
+///   loses no precision (both wire forms encode -9223372036854775808 exactly per the JSON
+///   Number spec). Using a non-strict `>= i64::MIN as f64` would let cases (b)/(c) into
+///   the i64 branch, but would also let case (a) silently saturate to `i64::MIN` — a
+///   silent data-corruption bug. The strict `>` is the safer trade-off.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Confirmed via RFC 8259 review: JSON itself does not mandate numeric precision (Section 6 explicitly defers to the implementation; IEEE-754 binary64 is mentioned only as a good-interop default). Fixed in commit f7ef2fe: the rustdoc now cites the actual two guarantees that make -2^63 round-trip exactly: (1) -2^63 is exactly representable as IEEE-754 binary64 (power-of-two within f64 normal range), and (2) serde_json serializes finite f64 values whose `fract() == 0.0` as exact decimal integer literals on the wire (no scientific notation, no decimal point). The new wording explicitly notes that the round-trip guarantee comes from f64 + serde_json semantics, not the JSON spec.

File: src/cli/issue/field_resolve.rs [hunk 8]
```diff
@@ -11,23 +11,101 @@ use crate::error::JrError;
...
+    let dec_part = &after_dot[1..]; // skip the '.'
+    if dec_part.is_empty() || !dec_part.chars().all(|c| c == '0') {
+        return None;
+    }
+    // Validate int_part: optional sign + ≥1 digit, all ASCII digits.
+    let digits = int_part.trim_start_matches(['-', '+']);
+    if digits.is_empty() || !digits.chars().all(|c| c.is_ascii_digit()) {
+        return None;
+    }
+    Some(int_part)
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Confirmed bug — `str::trim_start_matches` with a `&[char]` pattern strips ALL leading matching chars per the std docs, so `"--5.0"` was passing the digit check and returning `Some("--5")` instead of `None`. End-user impact was benign (parse failure cascade still produced the right error) but the helper violated its own `^[-+]?\d+\.0+$` contract.\n\nFixed in commit bbdfb8d: replaced the multi-tolerant trim with an explicit first-byte match (`Some(b'+') | Some(b'-') => &int_part[1..]`) that peels at most ONE sign. The `[1..]` slice is safe because `+` and `-` are single-byte ASCII so byte index 1 is always a valid char boundary when that arm is taken.\n\n4 new regression pins added to `test_strip_integer_decimal_suffix_rejects_malformed_input`: `"--5.0"`, `"++5.0"`, `"+-5.0"`, `"-+5.0"` all return None. Thanks for the careful read.

File: src/cli/issue/field_resolve.rs [hunk 9]
```diff
@@ -11,23 +11,107 @@ use crate::error::JrError;
...
+///
+/// This helper is called from the Stage 2 (f64) fallback in `resolve_edit_fields`.
+/// Stage 1 (`value.parse::<i64>()`) at the call site has already short-circuited all
+/// *string forms* that parse cleanly as i64 (e.g., plain integer literals like `"5"`
+/// or `"-9223372036854775807"`). Stage 2 therefore receives values from strings that
+/// failed i64 parsing — which includes:
+///
+/// - Decimal literals: `"5.5"` → emits f64; `"5.0"` → emits i64 via the fract predicate
+/// - Scientific notation: `"5e3"` → emits i64; `"1.5e3"` → emits i64; `"1e20"` → emits f64
+/// - Integer strings outside i64 range: `"9223372036854775808"` → emits f64
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit 54c0402: the Stage 2 caller-contract rustdoc example enumeration was a leftover from before the Stage 1.5 addition. Updated to list only inputs that ACTUALLY reach Stage 2 post-Stage-1.5: decimals with non-zero fractional digits (`"5.5"`, `"5.01"`), scientific notation (`"5e3"`, `"1e10"`, `"1e20"`), and integer strings WITHOUT decimal point that overflow i64 (`"9223372036854775808"`, `"-9223372036854775809"`). Added explicit cross-reference: inputs matching `^[-+]?\d+\.0+$` are intercepted by Stage 1.5 before reaching the helper. Thanks for catching the propagation gap.

File: src/cli/issue/field_resolve.rs [hunk 10]
```diff
@@ -11,23 +11,107 @@ use crate::error::JrError;
...
+///   (it's a power of two within f64's normal range), and (2) `serde_json` serializes
+///   finite f64 values whose `fract() == 0.0` as exact decimal integer literals on the
+///   wire (no scientific notation, no decimal point). Both
+///   `serde_json::Value::Number(Number::from(i64::MIN))` and
+///   `serde_json::json!(-9223372036854775808.0_f64)` produce the wire literal
+///   `-9223372036854775808`. JSON itself (RFC 8259) does not mandate numeric precision;
+///   the round-trip guarantee here is from f64 + `serde_json` semantics, not the JSON
+///   spec. Using a non-strict `>= i64::MIN as f64` would let cases (b)/(c) into the i64
+///   branch, but would also let case (a) silently saturate to `i64::MIN` — a silent
+///   data-corruption bug. The strict `>` is the safer trade-off.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: **Empirically verified — my claim was wrong.** Ran a direct serde_json serialization test:\n\n| Construct | Wire output |\n|---|---|\n| `Number::from(i64::MIN)` | `-9223372036854775808` (integer literal) |\n| `json!(-9223372036854775808.0_f64)` | `-9.223372036854776e+18` (**scientific notation**) |\n| `Number::from(5_i64)` | `5` |\n| `json!(5.0_f64)` | `5.0` (**decimal point, NOT bare integer**) |\n| `json!(5_i32)` | `5` |\n| `json!(5.5_f64)` | `5.5` |\n\nserde_json does NOT flatten integer-valued f64s to bare integer literals — it preserves f64-ness via Rust's default f64 Display. Your reference to tests 26-27 distinguishing `5` from `5.0` was exactly the structural test I should have written before making the claim.\n\nFixed in commit 54c0402:\n1. Removed the false serialization claim entirely.\n2. Replaced with empirically-verified wire-form descriptions (large-magnitude f64 → scientific notation; small whole f64 → decimal point; i64 path → bare integer literal).\n3. Corrected the case enumeration: case (b) \"decimal form of i64::MIN reaches Stage 2\" is wrong post-Stage-1.5 (it's intercepted and goes through the i64 wire path). Cases (a) underflow and (c) scientific notation are the actual Stage 2 entries with parsed = -2^63.\n4. Added explicit note that the integer-vs-float wire distinction IS observable by downstream consumers using NumericMode::Strict — which matters for tests 26/27 and any other strict-typed JSON consumers.

File: src/cli/issue/field_resolve.rs [hunk 11]
```diff
@@ -11,23 +11,128 @@ use crate::error::JrError;
...
+///   may arrive from several string forms:
+///
+///   - (a) An underflowing integer string like `"-9223372036854775809"` — Stage 1
+///     rejects it (parse fails); f64 rounds it to -2^63. Value is outside i64 range;
+///     emitting f64 is correct.
+///   - (b) The decimal form of `i64::MIN`: `"-9223372036854775808.0"` — Stage 1
+///     rejects it (decimal point). Value IS valid `i64::MIN`; strict `>` still routes
+///     it to f64.
+///   - (c) Scientific notation: `"-9.223372036854776e18"` — Stage 1 rejects it (`e`
+///     present). Value IS valid `i64::MIN` (approximately); strict `>` routes to f64.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs:53]: You're right — when I rewrote the rustdoc in R6 I added the 'Stage 1.5 intercepts decimal forms' note but left bullet (b) intact, creating a direct self-contradiction.\n\nFixed in commit cdc86fa: removed bullet (b) entirely. The actual Stage 2 source cases for `parsed == -2^63` are only (a) underflowing integer and (c) scientific notation. The 'intercepted by Stage 1.5' paragraph stays as the single source of truth for the decimal-form case.

File: src/cli/issue/field_resolve.rs [hunk 12]
```diff
@@ -496,4 +647,218 @@ mod tests {
...
+
+    #[test]
+    fn test_parsed_number_to_wire_value_strict_lower_excludes_negative_two_to_the_63_in_stage2() {
+        // -2^63 = i64::MIN as f64 (exact). In Stage 2 context, a parsed f64 value of
+        // -2^63 may arrive from several string forms: an underflowing integer like
+        // "-9223372036854775809", the decimal form of i64::MIN "-9223372036854775808.0",
+        // or scientific notation "-9.223372036854776e18". The strict > i64::MIN
+        // comparison routes ALL these cases to f64 regardless of source. This test
+        // invokes the helper directly with the f64 value -2^63 to pin the predicate
+        // behavior independent of source string.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Same stale enumeration mirrored in the test comment. Fixed in commit cdc86fa: comment updated to list only (a) underflow integer and (c) scientific notation as Stage 2 sources; decimal-form i64::MIN noted as intercepted by Stage 1.5. Test body unchanged.

File: src/cli/issue/field_resolve.rs [hunk 13]
```diff
@@ -11,23 +11,125 @@ use crate::error::JrError;
...
+///
+///   For case (a) — underflowing integer strings like `"-9223372036854775809"` — the
+///   value is outside i64 range; emitting f64 is correct. The wire form is scientific
+///   notation `-9.223372036854776e+18` (`serde_json` formats large-magnitude finite f64s
+///   using Rust's default f64 `Display`; it does NOT flatten integer-valued f64s to bare
+///   integer literals).
+///
+///   For case (c) — scientific notation `"-9.223372036854776e18"` — the value IS
+///   approximately `i64::MIN`, but the user supplied a non-integer string form; emitting
+///   f64 preserves that choice. Wire form is also scientific notation.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs]: Fixed in commit 2f8c24e: when R7 removed bullet (b) "decimal form of i64::MIN" from the Stage 2 lower-bound enumeration, I missed propagating to the prose that still cross-referenced "case (c)" (the scientific-notation case, which was renumbered to (b)). All remaining case-label references now match the actual (a)/(b) two-bullet enumeration.

File: src/cli/issue/field_resolve.rs [hunk 14]
```diff
@@ -496,4 +644,220 @@ mod tests {
...
+        } else if let Some(stripped) = super::strip_integer_decimal_suffix(value) {
+            if let Ok(n) = stripped.parse::<i64>() {
+                serde_json::Value::Number(serde_json::Number::from(n))
+            } else {
+                let parsed: f64 = value.parse().unwrap();
+                super::parsed_number_to_wire_value(parsed)
+            }
+        } else {
+            let parsed: f64 = value.parse().unwrap();
+            super::parsed_number_to_wire_value(parsed)
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs:716]: Good catch — you're right that the helper uses unwrap() + skips the is_finite() guard, so inputs like `"1e309"` (parses to +Inf) or `"NaN"` would panic via the helper's debug_assert! instead of going through production's user-error path.\n\nFixed in commit 2f8c24e: softened the helper's rustdoc to be explicit about its scope. Added a 'Limitations vs production' section documenting (a) unwrap() on f64 parse failures, (b) NaN/Inf inputs panic on the helper's debug_assert. Added guidance that tests needing NaN/Inf coverage should use HTTP-mocking end-to-end paths (current 6 tests use only finite valid inputs, so the gap is theoretical for them).\n\nKept the helper body unchanged because all current tests use it correctly; making the helper itself replicate the is_finite guard would just shift production's error path into the helper without meaningful coverage benefit.

File: src/cli/issue/field_resolve.rs [hunk 15]
```diff
@@ -11,23 +11,125 @@ use crate::error::JrError;
...
 pub(crate) fn parsed_number_to_wire_value(parsed: f64) -> serde_json::Value {
     debug_assert!(
         parsed.is_finite(),
         "parsed_number_to_wire_value requires a finite value; caller must reject NaN/Inf"
     );
-    if parsed.fract() == 0.0 && parsed >= i64::MIN as f64 && parsed <= i64::MAX as f64 {
+    if parsed.fract() == 0.0 && parsed > (i64::MIN as f64) && parsed < (i64::MAX as f64) {
         serde_json::Value::Number(serde_json::Number::from(parsed as i64))
     } else {
         serde_json::json!(parsed)
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/cli/issue/field_resolve.rs:99]: Good catch on the asymmetry — but this is the **F1 Option C design trade-off, explicitly accepted**, not a regression. The current code creates a 3-way asymmetry at exactly `±2^63`:\n\n| Input | Stage | Wire form |\n|---|---|---|\n| `"-9223372036854775808"` (i64::MIN literal) | Stage 1 i64 parse | integer `-9223372036854775808` |\n| `"-9223372036854775808.0"` (decimal form) | Stage 1.5 strip + retry | integer `-9223372036854775808` |\n| `"-9.223372036854776e18"` (scientific notation) | Stage 2 strict `>` rejects | f64 `-9.223372036854776e+18` |\n\nThe trade-off is documented in [the F1 delta analysis](https://github.com/Zious11/jira-cli/blob/factory-artifacts/phase-f1-delta-analysis/issue-421/delta-analysis.md) under "Fix Design Recommendation → Option C" — the strict `>` predicate prevents the silent saturation bug for the underflowing-integer case `"-9223372036854775809"` (which also parses to f64 -2^63), at the cost of routing valid-i64-value scientific notation at the boundary to f64 wire.\n\n**Test 27 contract check:** test 27 uses `"5e3"` (5000, well inside i64 range), not boundary scientific notation. The current code DOES preserve test 27's behavior — `"5e3"` → Stage 2 → f64 5000.0 → predicate `5000.0 > i64::MIN && 5000.0 < i64::MAX` → TRUE → emit i64 `5000`. The asymmetry is ONLY at the exact boundary value.\n\n**Why not pass `value: &str` into the helper?** That would let us distinguish `"-9.223372036854776e18"` from `"-9223372036854775809"` and emit i64 for the former. But:\n1. Inverts the helper's contract — it would become string-aware instead of f64-aware\n2. Requires Stage 2.5 routing logic to extract the integer-equivalent of scientific notation, validate it parses as i64, then emit i64. Substantial complexity for an exotic input class.\n3. User impact is essentially zero — who types i64::MIN in scientific notation to set a Jira number field?\n\n**Disposition:** Acknowledged as design intent. Documented in the rustdoc "case (b)" Stage 2 entry. If a future user genuinely needs boundary-scientific-notation as i64 wire form, that becomes a separate follow-up issue with the value/complexity discussion. For S-421's scope (eliminate the silent saturation bug at ±2^63), the current design is the right trade-off.\n\nThanks for the careful analysis — the asymmetry is real and worth flagging in the rustdoc, which it now is.
